### DISCUSSION TITLE
Generate relay-augmented schema for use with documentation generator

### DISCUSSION THREAD

**Discussion by seanf** [Netflix/dgs-framework#939] [Q&A] [answered]
Hi, I want to use the [relay pagination](https://netflix.github.io/dgs/advanced/relay-pagination/) feature (and thus the `@connection` directive), but I would still like to have a static schema, so that I can use the documentation generator [SpectaQL](https://github.com/anvilco/spectaql).

SpectaQL can be driven by a live GraphQL endpoint, a file containing a schema query introspection result, or a schema file (SDL). Up until now I've been pointing it at my schema file, but this won't work now because SpectaQL doesn't know about the `@connection` directive and the generated `*Connection` classes.

(I can't really think of a good way to give SpectaQL, in a Node module, live access to the DGS-generated schema, in a Java module. In any case, I don't think it would make things any simpler.)

Is there an easy way to get DGS to output the augmented schema (or introspection result) during my build?

**Comment by seanf** [comment, marked as answer]
In the end I wrote a class which uses `graphql.schema.idl.SchemaParser` to load a `TypeDefinitionRegistry`, ran it through `DgsPaginationTypeDefinitionRegistry.registry()` to generate the extra types, merged them with the original types, and then used `SchemaGenerator.makeExecutableSchema()` (with empty `RuntimeWiring`) and finally `SchemaPrinter` to write out the augmented SDL.

~~I also had to strip out the hash comments before building the `TypeDefinitionRegistry`, because the comments were converted to descriptions when printing the schema. See https://github.com/graphql-java/graphql-java/discussions/2765~~

I had to invoke `SchemaGenerator.makeExecutableSchema()` with `SchemaGenerator.Options.defaultOptions().useCommentsAsDescriptions(false)` to prevent hash comments from being turned into description strings. See https://github.com/graphql-java/graphql-java/discussions/2765#discussioncomment-2411123

**Comment by srinivasankavitha** [comment]
Thanks for updating this thread with your findings.
